### PR TITLE
Efa 3 refactors fixed

### PR DESCRIPTION
**Pull request by wzamazon**
3 EFA refactors

### CODE REVIEW

**Comment by wzamazon** [comment]
@shefty Can you share what intel CI failure is about? other tests are passing.

**Comment by shefty** [comment]
The failure is with the psm2 provider.

File: prov/efa/src/rxr/rxr_ep.c [hunk 1]
```diff
@@ -1730,7 +1730,8 @@ static size_t rxr_ep_post_rts(struct rxr_ep *rxr_ep, struct rxr_tx_entry *tx_ent
 
 	tx_entry->bytes_sent += data_sent;
 
-	if (!(efa_mr_cache_enable && tx_entry->total_len > data_sent))
+	if ((rxr_env.enable_shm_transfer && peer->is_local) ||
```
Comment (rajachan) [inline prov/efa/src/rxr/rxr_ep.c]: I must have missed this commit in the internal review.

This change works for the subsequent data packets. Are we doing the right thing for the RTS payload when sending over shm?

Comment (wzamazon) [reply prov/efa/src/rxr/rxr_ep.c]: shm does not require local memory registration, this patch skip this

Comment (rajachan) [reply prov/efa/src/rxr/rxr_ep.c]: Yep, I was asking more in the context of not copying/registering payload for shm when the cache is enabled, but I checked we do the right thing. We'll reuse the app buffer for the RTS payload with a change I will be making soon.

**Pull request merged**